web/src/app/util/SpeedDial.js

```diff
@@ -1,74 +1,70 @@
-import React from 'react'
+import React, { useState } from 'react'
 import p from 'prop-types'
-import SpeedDialAction from '@material-ui/lab/SpeedDialAction'
-import SpeedDialIcon from '@material-ui/lab/SpeedDialIcon'
-import SpeedDial from '@material-ui/lab/SpeedDial'
+import { SpeedDial, SpeedDialIcon, SpeedDialAction } from '@material-ui/lab'
+import { makeStyles } from '@material-ui/styles'
 
-export default class ScheduleNewOverrideFAB extends React.PureComponent {
-  static propsTypes = {
-    label: p.string.isRequired,
-    actions: p.arrayOf(
-      p.shape({
-        icon: p.element.isRequired,
-        onClick: p.func.isRequired,
-        label: p.string.isRequired,
-      }),
-    ).isRequired,
-  }
+const useStyles = makeStyles({
+  speedDial: {
+    position: 'fixed',
+    bottom: '2em',
+    right: '2em',
+    zIndex: 9001,
+  },
+  staticTooltipLabel: {
+    whiteSpace: 'nowrap',
+  },
+})
 
-  state = {
-    open: false,
-  }
+export default function CustomSpeedDial(props) {
+  const [open, setOpen] = useState(false)
+  const classes = useStyles()
 
-  shownState = false
-  _shownTimeout = -1
+  const handleToggle = () => setOpen(!open)
+  const handleOpen = () => setOpen(true)
+  const handleClose = () => setOpen(false)
 
-  render() {
-    if (this.state.open !== this.shownState) {
-      clearTimeout(this._shownTimeout)
-      this._shownTimeout = setTimeout(() => {
-        this.shownState = this.state.open
-      }, 350)
-    }
-    const doToggle = () => this.setState({ open: !this.shownState })
-    const doOpen = () => this.setState({ open: true })
-    const doClose = () => this.setState({ open: false })
-    return (
-      <SpeedDial
-        ariaLabel={this.props.label}
-        ButtonProps={{
-          'data-cy': 'page-fab',
-        }}
-        icon={<SpeedDialIcon />}
-        onClick={doToggle}
-        onClose={doClose}
-        onMouseEnter={doOpen}
-        onMouseLeave={doClose}
-        open={this.state.open}
-        style={{
-          position: 'fixed',
-          bottom: '2em',
-          right: '2em',
-          zIndex: 9001,
-        }}
-      >
-        {this.props.actions
-          .slice()
-          .reverse()
-          .map((action, idx) => (
-            <SpeedDialAction
-              key={idx}
-              icon={action.icon}
-              tooltipOpen
-              tooltipTitle={action.label}
-              aria-label={action.label}
-              onClick={() => {
-                this.setState({ open: false })
-                action.onClick()
-              }}
-            />
-          ))}
-      </SpeedDial>
-    )
-  }
+  return (
+    <SpeedDial
+      ariaLabel={props.label}
+      FabProps={{
+        'data-cy': 'page-fab',
+      }}
+      icon={<SpeedDialIcon />}
+      onClick={handleToggle}
+      onClose={handleClose}
+      onMouseEnter={handleOpen}
```

Comment:
I think we should be naming event handlers after the event, or just inlining the call. E.g. `handleClick` or `handleSpeedDialClick` (if there were multiple).

If we don't need specific event handlers (e.g. it's a simple function) we can just inline wth `() => setOpen(true)` instead.

Otherwise, I feel like the extra step is just adding noise.

Reply:
@Auchindoun thoughts?

Reply:
Getting around the eslint rule by using `() =>` seems like a way for us to slowly start avoiding the rule over time, which may lead to us just disabling it in the future so I would lean towards the current naming convention.

If we're only calling the set state hook function though, which is what's happening here, then I can see how it'd be better to just inline func it.

If it's using a function we've made that we're referencing though it should probably start with `handle`.